examples/gen-js/js/analytics/generated/index.d.ts

```diff
@@ -118,8 +118,25 @@ export interface ProfileViewed {
  * Analytics provides a strongly-typed wrapper around Segment Analytics
  * based on your Tracking Plan.
  */
+interface AnalyticsOptions {
+  onError: (
+    error: {
+      eventName: string;
+      validationErrors: Array<{
+        keyword: string;
+        dataPath: string;
+        schemaPath: string;
+        params: object;
+        message: string;
+        propertyName?: string;
+        parentSchema?: object;
+        data?: any;
+      }>;
```

Comment:
The error type comes from AJV: https://github.com/epoberezkin/ajv/blob/master/lib/ajv.d.ts#L279

To avoid a peer dependency on AJV, I just copied the type here.

Reply:
Would recommend we decompose this type a bit:


```typescript
interface AjvValidationError {
  keyword: string;
  dataPath: string;
  schemaPath: string;
  params: object;
  message: string;
  propertyName?: string;
  parentSchema?: object;
  data?: any;
}

interface InvalidEvent {
  eventName: string;
  validationErrors: AjvValidationError[];
}

interface AnalyticsOptions {
  onError(invalidEvent: InvalidEvent): void;
}
```

Reply:
Love it!